log/log.go

```diff
@@ -92,45 +92,37 @@ func newStdoutWriter(level int) *stdoutWriter {
 	}
 }
 
-// initialize is used in unit tests and called by `Get` before the logger
-// is fully set up.
-func initialize() {
-	const defaultPriority = syslog.LOG_INFO | syslog.LOG_LOCAL0
-	syslogger, err := syslog.Dial("", "", defaultPriority, "test")
-	if err != nil {
-		panic(err)
-	}
-	logger, err := New(syslogger, int(syslog.LOG_DEBUG), int(syslog.LOG_DEBUG))
-	if err != nil {
-		panic(err)
-	}
-
-	_ = Set(logger)
-}
-
-// Set configures the singleton Logger. This method
-// must only be called once, and before calling Get the
-// first time.
-func Set(logger Logger) (err error) {
+// Set configures the singleton Logger. This method must only be called once,
+// and before calling Get the first time.
+func Set(logger Logger) error {
```

Comment:
With so much of the functionality of this package being simplified I can't shake the feeling that both `Set()` and `Get()` being exported feels unnecessary. Ideally we should have just a couple of constructors that correctly initialize and return the logger we want. Perhaps there's something I'm missing?